src/main/scala/tech/cryptonomic/conseil/tezos/michelson/contracts/TokenContracts.scala

```diff
@@ -0,0 +1,143 @@
+package tech.cryptonomic.conseil.tezos.michelson.contracts
+
+import tech.cryptonomic.conseil.tezos.TezosTypes.{AccountId, Contract, ContractId, Decimal, ScriptId}
+import tech.cryptonomic.conseil.tezos.TezosTypes.Micheline
+import tech.cryptonomic.conseil.tezos.michelson.dto.{
+  MichelsonInstruction,
+  MichelsonIntConstant,
+  MichelsonSingleInstruction
+}
+import cats.implicits._
+import scala.util.Try
+import com.typesafe.scalalogging.LazyLogging
+import tech.cryptonomic.conseil.util.CryptoUtil
+
+/** Collects custom token contracts structures and operations */
+object TokenContracts extends LazyLogging {
```

Comment:
Now that I think about it, much of the logic in this class is not for all tokens but specifically for FA1.2 tokens. You should encapsulate the logic accordingly.

Reply:
I've though about this a couple of times already, and I'm more on the line of delaying such changes to when we need to integrate other specs.

Currently I think it will overcomplicate things and it might not eventually turn out useful, if the assumptions we make now are not validated once a new contract will be available.

This doesn't seem the last responsible moment for this choice. The solution is pretty simple now, and it will be better refactored once we have real information on the contracts to add and the related data-structures.